northstar/src/runtime/process/mod.rs

```diff
@@ -102,28 +111,15 @@ impl Launcher {
         // the waitpid of this child process signals that the child is exited. See
         // `wait`.
         let stop = CancellationToken::new();
-        let (init, argv) = init_argv(manifest, args);
+        let (cmd, argv) = init_argv(manifest, args);
         let mut env = self::env(manifest, env);
 
         // Setup io and collect fd setup set
         let (io, mut fds) = io::from_manifest(manifest).await?;
 
-        // Pipe for sending the init pid from the intermediate process to the runtime
-        // and the exit status from init to the runtime
-        //
-        // Ensure the fds of the channel are *not* in the fds set. The list of fds that are
-        // closed by init is gathered above. Between the assembly of the list and the new pipes
-        // for the child pid and the condition variables a io task that forwards logs from containers
-        // can end. Those io tasks use pipes as well. If such a task ends it closes its fds. Those numbers
-        // can be in the list of to be closed fds but are reused when the pipe are created.
-        let channel = channel::Channel::new();
-        fds.remove(&channel.as_raw_fd().0);
-        fds.remove(&channel.as_raw_fd().1);
-
-        // Ensure that the checkpoint fds are not in the fds set and untouched
-        let (checkpoint_runtime, checkpoint_init) = checkpoints();
-        fds.remove(&checkpoint_runtime.as_raw_fd().0);
-        fds.remove(&checkpoint_runtime.as_raw_fd().1);
+        // Two-way communication channels with the init process.
+        let mut comm_to_init = channel::Channel::<InitApi>::new();
+        let mut comm_from_init = channel::Channel::<InitMessage>::new();
```

Comment:
The file descriptors must be removed unconditionally from `fds`

Reply:
Yeah, I changed the `file_descriptor` member in `init.rs` to take a `skip` file descriptor list there. I decided to do it there since at this point in the code neither process has closed the unused ends.

Reply:
Got it.